src/java.base/share/classes/java/net/InetSocketAddress.java

```diff
@@ -119,7 +119,7 @@ public String toString() {
 
         @Override
         public final boolean equals(Object obj) {
-            if (obj == null || !(obj instanceof InetSocketAddressHolder))
+            if (!(obj instanceof InetSocketAddressHolder))
                 return false;
             InetSocketAddressHolder that = (InetSocketAddressHolder)obj;
```

Comment:
If we restructure this a little we can take advantage of flow scoping, e.g.

 public final boolean equals(Object obj) {
            if (!(obj instanceof InetSocketAddressHolder that))
                return false;
            boolean sameIP;
            if (addr != null)
                sameIP = addr.equals(that.addr);
            else if (hostname != null)
                sameIP = (that.addr == null) &&
                    hostname.equalsIgnoreCase(that.hostname);
            else
                sameIP = (that.addr == null) && (that.hostname == null);
            return sameIP && (port == that.port);
        }